front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -52,90 +52,101 @@ const ManchetteWithSpaceTimeChartWrapper = ({
-            );
-
-            if (!updatedSignalRange) return null;
-
-            // TODO : remove reformatting the datas when https://github.com/OpenRailAssociation/osrd-ui/issues/694 is merged
+
             return {
-              ...signal,
-              position_start: updatedSignalRange.spaceStart,
-              position_end: updatedSignalRange.spaceEnd,
-              time_start: updatedSignalRange.timeStart,
-              time_end: updatedSignalRange.timeEnd,
+              positions: cutPositions,
+              times: cutTimes,
             };
-          })
-        ),
-      }));
-
-      filteredConflicts = compact(
-        conflicts.map((conflict) => cutSpaceTimeRect(conflict, firstPosition, lastPosition))
-      );
+          }),
+          signalUpdates: compact(
+            train.signalUpdates.map((signal) => {
+              const updatedSignalRange = cutSpaceTimeRect(
+                {
+                  spaceStart: signal.position_start,
+                  spaceEnd: signal.position_end,
+                  timeStart: signal.time_start,
+                  timeEnd: signal.time_end,
+                },
+                firstPosition,
+                lastPosition
+              );
+
+              if (!updatedSignalRange) return null;
+
+              // TODO : remove reformatting the datas when https://github.com/OpenRailAssociation/osrd-ui/issues/694 is merged
+              return {
+                ...signal,
+                position_start: updatedSignalRange.spaceStart,
+                position_end: updatedSignalRange.spaceEnd,
+                time_start: updatedSignalRange.timeStart,
+                time_end: updatedSignalRange.timeEnd,
+              };
+            })
+          ),
+        }));
+
+        filteredConflicts = compact(
+          conflicts.map((conflict) => cutSpaceTimeRect(conflict, firstPosition, lastPosition))
+        );
+
+        return { filteredProjectPathTrainResult, filteredConflicts };
+      }
 
       return { filteredProjectPathTrainResult, filteredConflicts };
-    }
-
-    return { filteredProjectPathTrainResult, filteredConflicts };
-  }, [waypointsPanelData?.filteredWaypoints, projectPathTrainResult, conflicts]);
+    }, [waypointsPanelData?.filteredWaypoints, projectPathTrainResult, conflicts]);
+
+  const manchetteWaypoints = useMemo(() => {
+    const rawWaypoints = waypointsPanelData?.filteredWaypoints ?? operationalPoints;
+    return rawWaypoints.map((waypoint) => ({
+      id: waypoint.id,
+      position: waypoint.position,
+      name: waypoint.extensions?.identifier?.name,
+      secondaryCode: waypoint.extensions?.sncf?.ch,
+    }));
+  }, [waypointsPanelData, operationalPoints]);
 
   const { manchetteProps, spaceTimeChartProps, handleScroll } = useManchettesWithSpaceTimeChart(
-    waypointsPanelData?.filteredWaypoints ?? operationalPoints,
-    spaceTimeChartLayersData.filteredProjectPathTrainResult,
+    manchetteWaypoints,
+    cutProjectedTrains,
```

Comment:
Here we pass the projected trains with `signalUpdates` but the hook doesn't use them. Typescript doesn't seem to mind that there is an extra property in its wanted type but maybe we should remove it ?

Reply:
Yes, I think typescript only checks that the required props are here, it does not mind if extra props are given.
To remove the extra props, we will have to add a map. I think it's not very useful and we can stay like this, if it's good for you

Reply:
Fine by me